extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcIdentityProvider.java

```diff
@@ -62,7 +62,9 @@
             return Uni.createFrom().nullItem();
         }
         RoutingContext vertxContext = HttpSecurityUtils.getRoutingContextAttribute(request);
-        vertxContext.put(AuthenticationRequestContext.class.getName(), context);
+        if (request.getToken() instanceof IdTokenCredential) {
```

Comment:
@knutwannheden 

Thanks, I know I suggested that it was only required for `IdTokenCredential` but I think we should do it a bit differently.
I think we should start with
```
if (vertxContext == null && request.getToken() instanceof IdTokenCredential)  {
    return Uni.createFrom().failure(new AuthenticationFailedException());
}
```
and then continue with NPE guards, for ex, `OidcUtils` sets it as a `SecurityIdentity` attribute

Reply:
I will look into it.